### ISSUE TITLE
bad gateway

### ISSUE THREAD

**Issue by gald33** [danielwelch/hassio-zigbee2mqtt#453]
**zigbee2mqtt add-on version (if edge, please report commit hash):**
1.15.0

**Operating environment (HassOS, Virtual Machine, Device/platform running Home Assistant):**

**Description of problem:**
"Error 502: bad gateway" when I go to the addon (relative url: /7ad98f9c_zigbee2mqtt)

**more info:**

I used the same zigbee usb stick successfully with zigbee2mqtt (not the HA addon) a few days ago on another OS.
I added myself to the dialout group, and chown the device just in case.

What does this error mean?

Things I've tried:
- restarting the addon.
- waiting 20hrs.
- testing the usb dongle has write permission: after first command I connected the device
```
$ ls /dev/tty* >> ./temp/1.txt
$ ls /dev/tty* >> ./temp/2.txt
$ cd temp
$ diff 1.txt 2.txt 
65a66
> /dev/ttyACM0
$ test -w /dev/ttyACM0 && echo "write permission on" || echo "write permission off"
write permission on
```

**Your entire configuration from the frontend (with sensitive fields redacted)**:
```
haven't changed anything:

data_path: /share/zigbee2mqtt
external_converters: []
devices: devices.yaml
groups: groups.yaml
homeassistant: true
permit_join: false
mqtt:
  base_topic: zigbee2mqtt
  server: 'mqtt://core-mosquitto'
  user: my_user
  password: my_password
serial:
  port: /dev/ttyACM0
advanced:
  log_level: info
  pan_id: 6754
  channel: 11
  network_key:
    - 1
    - 3
    - 5
    - 7
    - 9
    - 11
    - 13
    - 15
    - 0
    - 2
    - 4
    - 6
    - 8
    - 10
    - 12
    - 13
  availability_blocklist: []
  availability_passlist: []
device_options: {}
blocklist: []
passlist: []
queue: {}
frontend:
  port: 8099
experimental:
  new_api: true
socat:
  enabled: false
  master: 'pty,raw,echo=0,link=/dev/ttyZ2M,mode=777'
  slave: 'tcp-listen:8485,keepalive,nodelay,reuseaddr,keepidle=1,keepintvl=1,keepcnt=5'
  restartdelay: 1
  initialdelay: 1
  options: '-d -d'
  log: false
```

**Your logs from Home Assistant**
Addon logs:
```
[s6-init] making user provided files available at /var/run/s6/etc...exited 0.
[s6-init] ensuring user provided files have correct perms...exited 0.
[fix-attrs.d] applying ownership & permissions fixes...
[fix-attrs.d] done.
[cont-init.d] executing container initialization scripts...
[cont-init.d] done.
[services.d] starting services
[services.d] done.
./run.sh: line 17: [Info] Configuration backup found in /share/zigbee2mqtt/.configuration.yaml.bk. Skipping config backup.: No such file or directory
[Info] Socat is DISABLED and not started
2020-10-27T09:35:52: PM2 log: Launching in no daemon mode
2020-10-27T09:35:52: PM2 log: App [npm:0] starting in -fork mode-
2020-10-27T09:35:52: PM2 log: App [npm:0] online
> zigbee2mqtt@1.15.0 start /zigbee2mqtt-1.15.0
> node index.js
Zigbee2MQTT:info  2020-10-27 09:35:52: Logging to console and directory: '/share/zigbee2mqtt/log/2020-10-27.09-35-52' filename: log.txt
Zigbee2MQTT:info  2020-10-27 09:35:52: Starting Zigbee2MQTT version 1.15.0 (commit #1.15.0)
Zigbee2MQTT:info  2020-10-27 09:35:52: Starting zigbee-herdsman...
Zigbee2MQTT:error 2020-10-27 09:36:33: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2020-10-27 09:36:33: Failed to start zigbee
Zigbee2MQTT:error 2020-10-27 09:36:33: Exiting...
Zigbee2MQTT:error 2020-10-27 09:36:33: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.<anonymous> (/zigbee2mqtt-1.15.0/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:92:31)
    at Generator.throw (<anonymous>)
    at rejected (/zigbee2mqtt-1.15.0/node_modules/zigbee-herdsman/dist/adapter/z-stack/adapter/zStackAdapter.js:25:65)
npm
 ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR!
 zigbee2mqtt@1.15.0 start: `node index.js`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the zigbee2mqtt@1.15.0 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Comment by ovizii** [comment]
I also had the same problem but it "magically" solved itself. I think it just takes really long for the addon to start and get ready as I didn't change anything, just checked again a few hours later. 

Maybe have a look if this thread helps? https://github.com/danielwelch/hassio-zigbee2mqtt/issues/446

**Comment by igiul667** [comment]
Did the ui ever work? It used to on my installation but then started giving this type of error. To fix it I reinstalled the addon

**Comment by gald33** [comment]
thanks @ovizii and @igiul667 , I've reinstalled the addon and waited for 20hrs, still getting the same error. After reading the thread you attached I wonder, do I need to install zigbee2mqtt beside zigbee2mqtt-addon?

**Comment by igiul667** [comment]
@gald33 I don't know how to help you then, I'm sorry. However I'm pretty much certain that you don't have to install zigbe2mqtt AND the addon, you need just one of the two.

**Comment by gald33** [comment]
thanks. maybe it's something in the configuration. I'll post if I find it.

**Comment by ciotlosm** [comment]
@gald33 You have multiple errors there:

`Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)` is your main problem. This means that zigbee2mqtt can't talk to your stick. Make sure that you use the correct serial port:

You currently have 
```
serial:
  port: /dev/ttyACM0
```

You might need to modify that and use the "named" serial port. You can view this based on the docs here: https://github.com/danielwelch/hassio-zigbee2mqtt/blob/master/zigbee2mqtt/DOCS.md#notes
`To find out which serial ports you have exposed go to Supervisor → System → Host system → ⋮ → Hardware`

`Error 502: bad gateway` is normal to see, because your add-on didn't start and the built-in frontend is not available, so `ingress` would just error out.

**Comment by gald33** [comment]
Thank you for all the answers. Solved. It turns out that there was an issue with the computer itself, and I had to replace the motherboard. It somehow effected the CC2531 and made it unusable, I think something might have been wrong with the computer voltage which could explain a bunch of problems I had the following day until it the computer finally crashed. Anyway, I re-flashed the CC2531 and then it got recognized again.
